lib/services/git-service.ts

```diff
@@ -89,77 +95,91 @@ export class GitService implements IGitService {
 	}
 
-	private async gitInit(projectDir: string): Promise<ISpawnResult> {
-		return this.executeCommand(projectDir, ["init"]);
+	private async gitInit(projectSettings: INSCloudProjectSettings): Promise<ISpawnResult> {
+		return this.executeCommand(projectSettings, ["init"]);
 	}
 
-	private async gitCommit(projectDir: string): Promise<ISpawnResult> {
-		return this.executeCommand(projectDir, ["commit", `-m "cloud-commit-${new Date().toString()}"`]);
+	private async gitCommit(projectSettings: INSCloudProjectSettings): Promise<ISpawnResult> {
+		return this.executeCommand(projectSettings, ["commit", `-m "cloud-commit-${new Date().toString()}"`]);
 	}
 
-	private async gitAdd(projectDir: string): Promise<ISpawnResult> {
-		return this.executeCommand(projectDir, ["add", projectDir]);
+	private async gitAdd(projectSettings: INSCloudProjectSettings): Promise<ISpawnResult> {
+		return this.executeCommand(projectSettings, ["add", projectSettings.projectDir]);
 	}
 
-	private async gitStatus(projectDir: string): Promise<ISpawnResult> {
-		return this.executeCommand(projectDir, ["status"]);
+	private async gitStatus(projectSettings: INSCloudProjectSettings): Promise<ISpawnResult> {
+		return this.executeCommand(projectSettings, ["status"]);
 	}
 
-	private async gitPush(projectDir: string, codeCommitCredential: ICodeCommitCredentials): Promise<ISpawnResult> {
+	private async gitPush(projectSettings: INSCloudProjectSettings, codeCommitCredential: ICodeCommitCredentials): Promise<ISpawnResult> {
 		const env = _.assign({}, process.env, {
 			AWS_ACCESS_KEY_ID: codeCommitCredential.accessKeyId,
 			AWS_SECRET_ACCESS_KEY: codeCommitCredential.secretAccessKey,
 			AWS_SESSION_TOKEN: codeCommitCredential.sessionToken
 		});
 
-		return this.executeCommand(projectDir, ["push", "--force", GitService.REMOTE_NAME, GitService.BRANCH_NAME], { env, cwd: projectDir });
+		return this.executeCommand(projectSettings, ["push", "--force", GitService.REMOTE_NAME, GitService.BRANCH_NAME], { env, cwd: projectSettings.projectDir });
 	}
 
-	private async gitRemoteAdd(projectDir: string, remoteUrl: IRemoteUrl, ) {
-		return this.executeCommand(projectDir, ["remote", "add", GitService.REMOTE_NAME, remoteUrl.httpRemoteUrl]);
+	private async gitRemoteAdd(projectSettings: INSCloudProjectSettings, remoteUrl: IRemoteUrl, ) {
+		return this.executeCommand(projectSettings, ["remote", "add", GitService.REMOTE_NAME, remoteUrl.httpRemoteUrl]);
+	}
+
+	private async gitSetRemoteUrl(projectSettings: INSCloudProjectSettings, remoteName: string, remoteUrl: IRemoteUrl) {
+		await this.executeCommand(projectSettings, ["remote", "set-url", remoteName, remoteUrl.httpRemoteUrl]);
+	}
+
+	private async gitCheckIfRemoteIsAdded(projectSettings: INSCloudProjectSettings, remoteName: string) {
+		try {
+			await this.executeCommand(projectSettings, ["remote", "get-url", remoteName]);
+			return true;
+		} catch (err) {
+			this.$logger.trace("Error while checking if remote is added: ", err);
```

Comment:
Do we need to log this as an error since this is the expected behavior if the remote is not set yet?

Reply:
This error is only in the debug logs. I decided to log it because it can be other error and we will be able to see only from the debug logs why this method returns false instead of true.